consul/src/main/scala/io/buoyant/consul/v1/BaseApi.scala

```diff
@@ -26,11 +26,17 @@ trait BaseApi extends Closable {
   private[this] val infiniteRetryFilter = new RetryFilter[http.Request, http.Response](
     RetryPolicy.backoff(backoffs) {
       // We will assume 5xx are retryable, everything else is not for now
-      case (_, Return(rep)) => rep.status.code >= 500 && rep.status.code < 600
+      case (req, Return(rep)) if rep.status.code >= 500 && rep.status.code < 600 =>
+        log.error(
+          s"Retrying Consul request '${req.method} ${req.uri}' on '${rep.status.code}' response"
```

Comment:
I'm torn on if we should include the `rep.contentString` as well here.  `ConsulApiError` does so maybe we should include that here as well.